no-vary-search.bs

```diff
@@ -189,6 +189,35 @@ The [=obtain a URL search variance=] algorithm ensures that all [=URL search var
   </table>
 </div>
 
+<div algorithm>
+  To <dfn>parse a key</dfn> given an [=ASCII string=] |keyString|:
+
+  1. Let |keyBytes| be the [=isomorphic encoding=] of |keyString|.
+
+  1. Replace any 0x2B (+) in |keyBytes| with 0x20 (SP).
+
+  1. Let |keyBytesDecoded| be the [=byte sequence/percent-decoding=] of |keyBytes|.
+
+  1. Let |keyStringDecoded| be the [=UTF-8 decode without BOM|UTF-8 decoding without BOM=] of |keyBytesDecoded|.
+
+  1. Return |keyStringDecoded|.
+</div>
+
+<div class="example" id="example-parse-a-key">
+  The [=parse a key=] algorithm allows encoding non-ASCII key strings in the ASCII strucured header format, similar to how the <a><code>application/x-www-form-urlencoded</code></a> format allows encoding an entire entry list of keys and values in ASCII URL format. For example,
+
+  <pre highlight="http">No-Vary-Search: params=("%C3%A9+%E6%B0%97")</pre>
+
+  will result in a [=URL search variance=] whose [=URL search variance/vary params=] are « "`é 気`" ». As explained in <a href="#example-equivalence-canonicalization">a later example</a>, the canonicalization process during [=equivalent modulo search variance|equivalence testing=] means this will treat as equivalent URL strings such as:
+
+  * `https://example.com/?é 気=1`
+  * `https://example.com/?é+気=2`
+  * `https://example.com/?%C3%A9%20気=3`
+  * `https://example.com/?%C3%A9+%E6%B0%97=4`
```

Comment:
Apart from the last one you shouldn't really see the first three at this layer though, right? Those can be input to `<a>`, but by the time we hit the caches and network they are normalized.

Reply:
Right, this relates to my below comment, and why I called these "URL strings" instead of URLs. Maybe it's too misleading though and we should overhaul all the examples to only operate on ASCII URL strings...